code/modules/hallucinations/effects/blind_rush_hallucination.dm

```diff
@@ -0,0 +1,69 @@
+/**
+  * # Hallucination - Blind Rush
+  *
+  * Makes target blind and causes them to see semi-transparent humanoids running at them.
+  */
+/datum/hallucination_manager/blind_rush
+	initial_hallucination = /obj/effect/hallucination/no_delete/blind_rusher
+	trigger_time = 34 //total length of the hallucination is a little more than ten seconds
+
+
+/datum/hallucination_manager/get_spawn_location()
+	var/list/turfs = range(13, owner.loc)
+	return pick(turfs)
+
+/datum/hallucination_manager/blind_rush/on_spawn()
+	owner.playsound_local(get_turf(src), istext('sound/spookoween/ghost_whisper.ogg') ? get_sfx('sound/spookoween/ghost_whisper.ogg') : 'sound/spookoween/ghost_whisper.ogg', 30, TRUE)
+	owner.become_blind("hallucination")
+	to_chat(owner, "<span class='whisper'>Who turned off the light?</span>", MESSAGE_TYPE_INFO)
+
+/datum/hallucination_manager/on_trigger()
+	var/turf/spawn_location = get_spawn_location() //we need a new spawn location incase the player moved
+	new /obj/effect/hallucination/no_delete/blind_rusher(spawn_location, owner)
+	to_chat(owner, "<span class='danger'>They're here.</span>", MESSAGE_TYPE_INFO)
+	trigger_timer = addtimer(CALLBACK(src, PROC_REF(on_second_trigger)), trigger_time, TIMER_DELETE_ME)
+
+/datum/hallucination_manager/proc/on_second_trigger()
+	var/turf/spawn_location = get_spawn_location()
+	new /obj/effect/hallucination/no_delete/blind_rusher(spawn_location, owner)
+	owner.Confused(9 SECONDS)
+	owner.Jitter(9 SECONDS)
+	to_chat(owner, "<span class='userdanger'>Run!</span>", MESSAGE_TYPE_INFO)
+	trigger_timer = addtimer(CALLBACK(src, PROC_REF(on_last_trigger)), trigger_time, TIMER_DELETE_ME)
+
+/datum/hallucination_manager/proc/on_last_trigger()
+	owner.emote("collapse")
+	owner.cure_blind("hallucination")
+	qdel(src)
+
+/obj/effect/hallucination/no_delete/blind_rusher
+	alpha = 100
+	hallucination_plane = 25 //to make sure we render the hallucination above the blindness layer.
+	hallucination_layer = 25
+	hallucination_icon = 'icons/mob/simple_human.dmi'
+	hallucination_icon_state = ("faceless")
+	hallucination_override = TRUE
+	var/min_distance = 0
+	var/rush_time = 2
+	var/rush_timer = null
+
+/obj/effect/hallucination/no_delete/blind_rusher/Initialize(mapload, mob/living/carbon/target)
+	name = "Unknown"
+	rush_timer = addtimer(CALLBACK(src, PROC_REF(rush)), rush_time, TIMER_LOOP | TIMER_STOPPABLE)
```

Comment:
```suggestion
	rush_timer = addtimer(CALLBACK(src, PROC_REF(rush)), rush_time, TIMER_LOOP | TIMER_STOPPABLE | TIMER_DELETE_ME)
```

Reply:
Won't this mean the two other instance of the hallucination won't have a timer assigned to them after it deleted itself?

Reply:
well, since doing this runtimes, DGamer has recommended I resolve this review (and the one about the destroy() call aswell) despite not applying the changes